packages/compass-query-history/src/stores/recent-list-store.js

```diff
@@ -71,7 +71,7 @@ const configureStore = (options = {}) => {
       });
 
       /* Ignore duplicate queries */
-      const existingQuery = this.state.items.find((item) =>
+      const existingQuery = filtered.find((item) =>
```

Comment:
Drive-by: When we filtered duplicate queries we didn't take the namespace into account. So if you've run filter `{ foo: 1 }` against collection A and now run it against B it wouldn't add it to B's recent query list. Oops.